### PR TITLE
Basic socket emit wrapping

### PR DESCRIPTION
**Pull request by achandras**

### CODE REVIEW

File: src/plugins/socket.io.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      try {
+        retVal = scope.bind(emit, span).call(this, eventName, callback)
+      } catch (e) {
+        if (span) {
+          span.addTags({
+            'error': true,
+            'message': e.message,
+            'stack': e.stack,
+            'error.kind': e.name
+          })
```
Comment (rmfitzpatrick) [inline src/plugins/socket.io.js]: `throw e`?

File: test/plugins/socket.io.spec.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  transports: ['websocket'],
+  forceNew: true,
+  reconnection: false
+}
+
+const express = require('express')
+
+wrapIt()
+
+function setupApp (module) {
```
Comment (rmfitzpatrick) [inline test/plugins/socket.io.spec.js]: nit: If `module` is always an imported socket.io maybe something like `sio`?

**Comment by rmfitzpatrick** [comment]
What led you to not use the scope manager's `bind()` compatibility w/ event emitters?  Also, wouldn't tracing listeners be as helpful as emitted events?

**Comment by achandras** [comment]
I can add a `bind()` using the emitter. I had a limited understanding of how that could be useful in this context, given that the `emit` call is already binding the scope. Do you have any suggestions how it could be integrated better?

For listeners, I had initially considered tracing them, but couldn't find much useful information to put other than `listener funcName` if it wasn't an anonymous function, and maybe just `listener` or `listener anonymous` if it was. If that would in fact be useful, I can add it in. My thought was that any useful spans would have to come from instrumentation in the listener, which would be correctly nested with the function `bind`.

File: src/plugins/socket.io.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        'namespace': this.name,
+        'path': this.server._path,
+        'clients.count': this.server.eio.clientsCount,
+        'events.count': this._eventsCount
+      }
+
+      const span = tracer.startSpan('emit ' + eventName, { childOf, tags: tags })
+
+      try {
+        scope.bind(this, span)
```
Comment (rmfitzpatrick) [inline src/plugins/socket.io.js]: What is this intended to do?

**Comment by rmfitzpatrick** [comment]
> Do you have any suggestions how it could be integrated better?

For posterity, discussed this offline and my recommendation was for `bind()`'s event emitter wrapping functionality and not its relationship with an active span.  The current implementation seems sound.

File: src/plugins/socket.io.js (hunk 3 above)
Comment (achandras) [reply src/plugins/socket.io.js]: Sorry, leftovers from earlier experimentation. Removed

**Pull request merged**